### PR TITLE
ORCH-456 Fix version parsing

### PR DESCRIPTION
**Pull request by malena-ebert-sonarsource**

### CODE REVIEW

**Review by rose-goullet-sonarsource — COMMENTED** [review]
The logic LGTM!
These code smells should probably be fixed though, and I think lines 54-55 can be covered by trying to create a version from a purely alphabetical string. Also I'm not sure if line 64 (no major version number found) can ever be reached?

File: sonar-orchestrator/src/main/java/com/sonar/orchestrator/version/Version.java [hunk 1]
```diff
@@ -19,43 +19,76 @@
...
+import org.jetbrains.annotations.Nullable;
+
 import java.util.Objects;
-import org.apache.commons.lang.StringUtils;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 public class Version implements Comparable<Version> {
 
+  private static final String GROUP_MAJOR        = "major";
```
Comment (jacek-poreda-sonarsource) [inline sonar-orchestrator/src/main/java/com/sonar/orchestrator/version/Version.java:30]: although for sure it is more readable, but unfortunately it does not follow our formatting guide, see https://github.com/SonarSource/sonar-developer-toolset

**Review by jacek-poreda-sonarsource — COMMENTED** [review]
There are still some easy to fix code smells and coverage to be improved. I think it should be easy to cover it 100% as class is very small.

Also maybe it could be worth to update `VersionTest` class. On my end I see that `SonarLint` highlight more to be fixed for instance: `isEqualTo` instead of `assertThat(...).equals(...).isTrue();`.  Could be good opportunity to clean things a little :)

**Comment by malena-ebert-sonarsource** [comment]
> I think it should be easy to cover it 100%

This is not possible that the major group is null.

**Review by jacek-poreda-sonarsource — APPROVED** [review]

**Pull request merged**